### ISSUE TITLE
Multiple iFrames spawned in case of multiple silent token fetch with refresh token expired

### ISSUE THREAD

**Issue by rhnsharma999** [AzureAD/microsoft-authentication-library-for-js#2163]
## Library
- [ ] `msal@1.x.x` or `@azure/msal@1.x.x`
- [x] `@azure/msal-browser@2.x.x`
- [ ] `@azure/msal-angular@0.x.x`
- [ ] `@azure/msal-angular@1.x.x`
- [ ] `@azure/msal-angularjs@1.x.x`

## Description
Hi MSAL team.
We are in the process of migrating our existing auth flow to MSAL 2.0

As as a part of the authentication flow, we fetch multiple access tokens.
We saw a weird behavior and wanted to get that clarified.

We try to acquire tokens silently using `Promise.all([])` calling `myMSALObj.acquireTokenSilent()`
```typescript
Promise.all([myMSALObj.acquireTokenSilent(silentRequest1), myMSALObj.acquireTokenSilent(silentRequest2)])
``` 
Following are the assumptions
- The user already logged in at some point and tokens are cached in the localStorage.
- The refresh token is now expired.

As we understood from the documentation, when the refresh token is expired, MSAL tries to spawn an iFrame to fetch the new tokens. 
In our it does that as well but spawns more than one iFrame.

So if we try to fetch 2 tokens, silently, 2 iFrames are spawned (1 for each call to `acquireTokenSilent`)

Questions:
- Is this the correct way of fetching multiple tokens?
- Is there a way to prevent another iFrame from being spawned when one is already present?
- I believe this can be prevented if we make calls to `acquireTokenSilent` sequentially and only when the previous one is resolved, but that would be very slow.

Your help will be greatly appreciated.

here is our MSAL config
```typescript
export const MSAL_CONFIG: Configuration = {
  auth: {
    clientId: '<our client id here>',
    authority: 'https://login.windows-ppe.net/common',
  },
  cache: {
    cacheLocation: 'localStorage', // This configures where your cache will be stored
    storeAuthStateInCookie: false, // Set this to "true" if you are having issues on IE11 or Edge
  },
  system: {
    loggerOptions: {
      loggerCallback: (level, message, containsPii) => {
        if (containsPii) {
          return
        }
        switch (level) {
          case LogLevel.Error:
            console.error(message)
            return
          case LogLevel.Info:
            console.info(message)
            return
          case LogLevel.Verbose:
            console.debug(message)
            return
          case LogLevel.Warning:
            console.warn(message)
            return
        }
      },
    },
  },
}
```

**Comment by technical-boy** [comment]
Hi, @rhnsharma999! 

Are you encountering any errors or broken behavior in this scenario? Otherwise, do you have any specific concerns with MSAL spawning multiple iframes? If that's the case, what behavior would you expect when calling multiple `acquireTokenSilent`s concurrently?

We'd appreciate it if you could complete the issue template so we can have enough information to answer your question.

**Comment by rhnsharma999** [comment]
Hi @technical-boy 
I don't see any issues or weird behavior while multiple iFrames are spawned.
The concern that i have is that for each `acquireTokenSilent` call we'll have to wait for the respective iFrame to finish. And in case one of them fails (and assuming at least one other succeeds), then error handling would be complex as well. (I know this is an extreme edge case but it might happen) 

Ideally I would expect only one iFrame to be spawned (since its fetching only type of token) and any subsequent calls to `acquireTokenSilent` would actually wait for iFrame to finish. 

And I am sorry, I used the entire `Question Template`, not sure what other info is missing.

**Comment by technical-boy** [comment]
Hey @rhnsharma999 . In regards to error handling, you can use [Promise.allSettled](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/allSettled) instead of `Promise.all`, which will always return a result promise even if one of the promises is rejected.

If you are concerned with the performance or side effects of multiple iframes, you should perform these requests in serial. We're looking into ways we can improve the behavior of refresh token renewal to address your concerns.

**Comment by rhnsharma999** [comment]
Hi @technical-boy,
We have indeed started seeing issues with the multiple iFrame spawn. This issue is consistently reproducible.

I am adding the details below


## Library
- [ ] `msal@1.x.x` or `@azure/msal@1.x.x`
- [x] `@azure/msal-browser@2.x.x`
- [ ] `@azure/msal-angular@0.x.x`
- [ ] `@azure/msal-angular@1.x.x`
- [ ] `@azure/msal-angularjs@1.x.x`

`@azure/msal-browser@2.0.1`

## Framework
N/A

## Description
Parallel `acquireTokenSilent` calls fail when the refresh token is expired.

## Error Message
```
ClientAuthError: state_mismatch: State mismatch error. Please check your network. Continued requests may cause cache overflow.
    at ClientAuthError.AuthError [as constructor] (http://localhost:3000/static/js/0.chunk.js:450:20)
    at new ClientAuthError (http://localhost:3000/static/js/0.chunk.js:730:24)
    at Function.ClientAuthError.createStateMismatchError (http://localhost:3000/static/js/0.chunk.js:813:12)
    at ResponseHandler.validateServerAuthorizationCodeResponse (http://localhost:3000/static/js/0.chunk.js:4058:29)
    at AuthorizationCodeClient.handleFragmentResponse (http://localhost:3000/static/js/0.chunk.js:4521:21)
    at SilentHandler.<anonymous> (http://localhost:3000/static/js/0.chunk.js:7614:40)
    at step (http://localhost:3000/static/js/0.chunk.js:198:17)
    at Object.next (http://localhost:3000/static/js/0.chunk.js:129:14)
    at http://localhost:3000/static/js/0.chunk.js:101:67
    at new Promise (<anonymous>)
    at __awaiter (http://localhost:3000/static/js/0.chunk.js:78:10)
    at SilentHandler.InteractionHandler.handleCodeResponse (http://localhost:3000/static/js/0.chunk.js:7603:12)
    at PublicClientApplication.<anonymous> (http://localhost:3000/static/js/0.chunk.js:8758:29)
    at step (http://localhost:3000/static/js/0.chunk.js:198:17)
    at Object.next (http://localhost:3000/static/js/0.chunk.js:129:14)
    at fulfilled (http://localhost:3000/static/js/0.chunk.js:81:24)
```

## Security
- Issue is **not** security related

## Regression
- It **did not** work before
Version:

## MSAL Configuration
This is the same as the typescript sample. No change has been made
```js
export const MSAL_CONFIG: Configuration = {
  auth: {
    clientId: '2eb9245f-612b-46cc-994a-f5e35ef37da0',
  },
  cache: {
    cacheLocation: 'localStorage', // This configures where your cache will be stored
    storeAuthStateInCookie: false, // Set this to "true" if you are having issues on IE11 or Edge
  },
  system: {
    loggerOptions: {
      loggerCallback: (level, message, containsPii) => {
        if (containsPii) {
          return
        }
        switch (level) {
          case LogLevel.Error:
            console.error(message)
            return
          case LogLevel.Info:
            console.info(message)
            return
          case LogLevel.Verbose:
            console.debug(message)
            return
          case LogLevel.Warning:
            console.warn(message)
            return
        }
      },
    },
    windowHashTimeout: 30000,
    iframeHashTimeout: 30000,
  },
}
```
## Context
In our web app flow, we don't make a `myMSALObj.loginRedirect()` call every time the app loads. Instead we rely on the `myMSALObj.getAllAccounts()` to see if a user has logged in or not. 

If `myMSALObj.getAllAccounts()` returns null, we redirect for login (the usual way).

Then we make parallel `acquireTokenSilent()` calls for multiple resources (3 to be exact) using `Promise.all()`

## Reproduction steps
When the `accessTokens` and the `refreshToken` have expired, parallel `acquireTokenSilent()` calls fail with the above mentioned error.  

## Expected behavior
The respective calls should succeed.

## Browsers/Environment
- [x] Chrome
- [x] Firefox
- [x] Edge
- [x] Safari
- [x] IE
- [ ] Other (Please add browser name here)

**Comment by jasonnutter** [comment]
This will be addressed in #2290
